src/server/services/penalty.service.js

```diff
@@ -44,9 +44,31 @@ export default class PenaltyService {
     return penaltyDetails;
   }
 
+  static parsePayments(paymentsArr) {
+    const splitAmounts = paymentsArr.map((payment) => { // eslint-disable-line arrow-body-style
+      return {
+        type: payment.PaymentCategory,
+        amount: payment.TotalAmount,
+        status: payment.PaymentStatus,
+      };
+    });
+    const types = uniq(paymentsArr.map(payment => payment.PaymentCategory));
+    const parsedPenalties = types.map((type) => {
+      const penalties = paymentsArr.filter(p => p.PaymentCategory === type)[0].Penalties;
+      return {
+        type,
+        penalties: penalties.map(p => PenaltyService.parsePenalty(p)),
+      };
+    });
+    const unpaidPayments = paymentsArr.filter(payment => payment.PaymentStatus === 'UNPAID');
+    const fpnPayment = find(unpaidPayments, ['PaymentCategory', 'FPN']);
+    const nextPayment = typeof fpnPayment !== 'undefined' ? fpnPayment : find(unpaidPayments, ['PaymentCategory', 'IM']);
```

Comment:
This doesn't take court deposit (`CDN`) into account. Probably refactor to another function.